docs/file-reference/index.md

```diff
@@ -0,0 +1,642 @@
+| events                                                           | [eventObject](#event-object)                | IN PROGRESS | Refer to issue: https://github.com/openshift/odo/issues/2919                                    |
+| component[].kubernetes                                           | [kubernetesObject](#kubernetesobject)       | IN PROGRESS |                                                                                                 |
+| component[].openshift                                            | [openshiftObject](#openshiftobject)         | IN PROGRESS |                                                                                                 |
+| component[].volume                                               | [volumeObject](#volumeobject)               | IN PROGRESS | Refer to: https://github.com/openshift/odo/issues/3407                                          |
+| component[].plugin                                               | [pluginObject](#pluginobject)               | IN PROGRESS | Refer to: https://github.com/openshift/odo/issues/3407                                          |
+| component[].container.endpoints.configuration.cookiesAuthEnabled | [configurationObject](#configurationobject) | UNKNOWN     | In schema but not yet implemented.                                                              |
+| component[].container.endpoints.configuration.type               | [configurationObject](#configurationobject) | UNKNOWN     | In schema but not yet implemented.                                                              |
+| commands[].apply                                                 | [applyObject](#applyobject)                 | UNKNOWN     | In schema but not yet implemented.                                                              |
+| commands[].composite                                             | [compositeObject](#compositeobject)         | IN PROGRESS | Refer to: https://github.com/openshift/odo/issues/3338                                          |
+| commands[].vscodeLaunch                                          | [vscodeLaunchObject](#vscodeLaunchobject)   | N/A         | Not applicable to odo.                                                                          |
+| commands[].vscodeTask                                            | [vscodeTaskObject](#vscodeTaskobject)       | N/A         | Not applicable to odo.                                                                          |
+
+
+
+## Installation
+
+Installing odo can be found at our [installation page](https://odo.dev/docs/installing-odo/).
+
+## Quick Start
+
+If you haven't used odo yet, we recommend going through our [Deploying a devfile using odo guide](https://odo.dev/docs/deploying-a-devfile-using-odo/). 
+
+
+# Devfile Properties
```

Comment:
Wouldn't be better to just link to official devfile reference?

Reply:
Agreed, as it stands, this is more a devfile doc which should probably live under the devfile project as it doesn't really have much information pertaining to what can be done with odo and devfiles…